app/models/event.rb

```diff
@@ -284,5 +241,70 @@ def time_detail(options = {})
     return "#{_start_date}#{separator[:at]} #{_start_time}#{separator[:to]} #{_end_date}#{separator[:at]} #{_end_time}"
   end
 
+  # Returns whether or not it is too close to the event to cancel RSVPs
+  # for the given audience, if it was set by the +event_group+
+  def rsvps_disabled?(person_or_type)
+    return false unless event_group
+    hours_prior = event_group.hours_prior_disable_cancel(person_or_type)
+    return false if hours_prior.blank?
+    ((start_datetime.to_time - Time.now) / 1.hour) <= hours_prior
+  end
+
+  # Takes one argument for which we attempt to determine the associated
+  # auidience: one of Mentor, Volunteer, Student, or Participant
+  #   * A string or symbol matching one of the above audiences
+  #     (case-insensitive)
+  #   * An ActiveRecord object of one of the above 4 types
+  #   * One of the 4 above classes as a constant e.g. `Mentor`
+  #
+  # Raises ArgumentError on nil, as checks for nil audience should
+  # be handled by the caller before invoking this method. Will
+  # also raise ArgumentError if person_or_type cannot be resolved
+  # to an audience.
+  #
+  # Returns a symbol representing the audience, e.g. `:Mentor`
+  def self.process_audience(person_or_type)
+    unless person_or_type.is_a?(Symbol)
+      person_or_type = if person_or_type.is_a?(Person)
+                         person_or_type.class.to_s
+                       elsif person_or_type.is_a?(String)
+                         person_or_type
+                       elsif person_or_type.is_a?(Class)
+                         person_or_type.to_s
+                       else
+                         raise ArgumentError, "Cannot parse audience", caller
+                       end
+
+      person_or_type = person_or_type.to_sym
+    end
+
+    person_or_type = person_or_type.capitalize
+    unless %i(Volunteer Mentor Student Participant).include? person_or_type
+      raise ArgumentError, "Invalid audience type", caller
+    end
+
+    person_or_type
+  end
+
+  protected
+
+  # Helper methods for returning attributes prefixed by audience
+
+  # Use this if there is a generic, non-audience attribute
+  # which is the default
+  def attribute_for_audience_with_generic(attr, person_or_type)
+    generic_attribute = read_attribute(attr)
+    return generic_attribute if person_or_type.nil?
+    aud = Event.process_audience(person_or_type)
+    custom_attribute = attribute_for_audience(attr, aud)
+    custom_attribute.blank? ? generic_attribute : custom_attribute
+  end
+
+  def attribute_for_audience(attr,  audience)
```

Comment:
@mattharris5 I defined the exact same method in EventGroup. It still reduces overall complexity, but the duplication is making CodeClimate say the event_group.rb has gotten worse (which I guess it technically has). The OOP thing would be to make them both inherit from AudienceScoped or something, but I don't know if that's worth it.